src/DetailsView/components/settings-panel/settings/bug-filing/bug-filing-settings.tsx

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { TextField } from 'office-ui-fabric-react/lib/TextField';
+import * as React from 'react';
+
+import { FlaggedComponent } from '../../../../../common/components/flagged-component';
+import { FeatureFlags } from '../../../../../common/feature-flags';
+import { UserConfigMessageCreator } from '../../../../../common/message-creators/user-config-message-creator';
+import { NamedSFC } from '../../../../../common/react/named-sfc';
+import { FeatureFlagStoreData } from '../../../../../common/types/store-data/feature-flag-store-data';
+import { UserConfigurationStoreData } from '../../../../../common/types/store-data/user-configuration-store';
+
+export type BugFilingSettingsDeps = {
+    userConfigMessageCreator: UserConfigMessageCreator;
+};
+export type BugFilingSettingsProps = {
+    deps: BugFilingSettingsDeps;
+    featureFlagData: FeatureFlagStoreData;
+    userConfigStoreState: UserConfigurationStoreData;
+};
+
+export const BugFilingSettings = NamedSFC<BugFilingSettingsProps>('BugFilingSettings', props => {
+    const getBugSettingsUx = () => {
+        return (
+            <>
+                <h3>Issue filing</h3>
+                {getGitHubBugSettingsUx()}
+            </>
+        );
+    };
+
+    const getGitHubBugSettingsUx = () => {
+        return (
+            <TextField
+                className="issue-setting"
+                label="Enter desired GitHub repo link:"
+                onChange={onGitHubRepositoryChange}
+                value={getBugServiceProperty('gitHub', 'repository')}
+                placeholder="https://github.com/owner/repo/issues"
+            />
+        );
+    };
+
+    const onGitHubRepositoryChange = (event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, newValue?: string) => {
```

Comment:
Are these methods not complaining of missing return type?

Reply:
They are not... I was playing with typescript version the other day and 3.x complains about this... not sure why 2.x does not